### ISSUE TITLE
python3.11如何进行VLLM部署

### ISSUE THREAD

**Issue by shuyanlei** [OpenDCAI/Paper2Any#154]
求问为什么readme里给的教程，建立虚拟环境时用的是python3.11，但是后面可选择vllm部署连接里只有python3.12的版本？python3.11该怎么部署vllm呢？

**Comment by DeepThinkingZhouLiu** [comment]
你好，这里的关键点是 wheel 文件名中的 Python ABI 必须和环境版本一致。

文档示例中的：

```text
vllm-0.11.0+cu124-cp312-cp312-win_amd64.whl
```

其中 `cp312` 表示它只适用于 CPython 3.12，不能直接安装到 Python 3.11 环境中。强行在 3.11 环境安装会因为 ABI 不匹配而失败。

可以选择下面两种方案。

**方案一：Windows 下统一使用 Python 3.12**

当前 README 的 Windows 安装章节已经统一调整为 Python 3.12，这是原生 Windows 下最省事的方案：

```bash
conda create -n paper2any python=3.12 -y
conda activate paper2any
```

然后安装与 `cp312`、Windows 架构和本机 CUDA 版本匹配的 vLLM wheel。

**方案二：Paper2Any 和 vLLM 使用两个独立环境**

Paper2Any 主程序可以继续运行在 Python 3.11，vLLM 单独使用 Python 3.12 环境：

```bash
conda create -n paper2any python=3.11 -y
conda create -n paper2any-vllm python=3.12 -y
```

在 `paper2any-vllm` 环境中安装对应的 `cp312` wheel，并启动 MinerU 模型服务：

```bash
conda activate paper2any-vllm
vllm serve opendatalab/MinerU2.5-2509-1.2B \
  --host 127.0.0.1 \
  --port 8010 \
  --logits-processors mineru_vl_utils:MinerULogitsProcessor \
  --trust-remote-code
```

Paper2Any 与 vLLM 之间通过 HTTP 通信，并不要求在同一个 Python 环境中。主程序侧保持：

```env
MINERU_PORT=8010
```

即可访问本机的 vLLM/MinerU 服务。如果使用 Linux 或 WSL，也可以在独立环境中安装官方支持 Python 3.11 的 vLLM 版本，但仍建议把模型服务环境与 Paper2Any 主环境隔离，减少 PyTorch、CUDA 和其他依赖之间的冲突。

因此，Python 3.11 不能直接使用示例里的 `cp312` Windows wheel，但可以通过独立 Python 3.12 模型服务环境解决。当前文档已明确 Windows 使用 Python 3.12，这条问题按已说明关闭；如仍有具体安装错误，欢迎附上操作系统、Python、CUDA、显卡驱动版本以及所用 wheel 文件名重新打开。

**Comment by shuyanlei** [comment]
谢谢！







-----原始邮件-----
发件人:"Zhou Liu" ***@***.***>
发送时间:2026-08-03 17:22:31 (星期一)
收件人: OpenDCAI/Paper2Any ***@***.***>
抄送: emolake ***@***.***>, Author ***@***.***>
主题: Re: [OpenDCAI/Paper2Any] python3.11如何进行VLLM部署 (Issue #154)


DeepThinkingZhouLiu left a comment (OpenDCAI/Paper2Any#154)

你好，这里的关键点是 wheel 文件名中的 Python ABI 必须和环境版本一致。

文档示例中的：

vllm-0.11.0+cu124-cp312-cp312-win_amd64.whl


其中 cp312 表示它只适用于 CPython 3.12，不能直接安装到 Python 3.11 环境中。强行在 3.11 环境安装会因为 ABI 不匹配而失败。

可以选择下面两种方案。

方案一：Windows 下统一使用 Python 3.12

当前 README 的 Windows 安装章节已经统一调整为 Python 3.12，这是原生 Windows 下最省事的方案：

conda create -n paper2any python=3.12 -y
conda activate paper2any

然后安装与 cp312、Windows 架构和本机 CUDA 版本匹配的 vLLM wheel。

方案二：Paper2Any 和 vLLM 使用两个独立环境

Paper2Any 主程序可以继续运行在 Python 3.11，vLLM 单独使用 Python 3.12 环境：

conda create -n paper2any python=3.11 -y
conda create -n paper2any-vllm python=3.12 -y

在 paper2any-vllm 环境中安装对应的 cp312 wheel，并启动 MinerU 模型服务：

conda activate paper2any-vllm
vllm serve opendatalab/MinerU2.5-2509-1.2B \
  --host 127.0.0.1 \
  --port 8010 \
  --logits-processors mineru_vl_utils:MinerULogitsProcessor \
  --trust-remote-code

Paper2Any 与 vLLM 之间通过 HTTP 通信，并不要求在同一个 Python 环境中。主程序侧保持：

MINERU_PORT=8010

即可访问本机的 vLLM/MinerU 服务。如果使用 Linux 或 WSL，也可以在独立环境中安装官方支持 Python 3.11 的 vLLM 版本，但仍建议把模型服务环境与 Paper2Any 主环境隔离，减少 PyTorch、CUDA 和其他依赖之间的冲突。

因此，Python 3.11 不能直接使用示例里的 cp312 Windows wheel，但可以通过独立 Python 3.12 模型服务环境解决。当前文档已明确 Windows 使用 Python 3.12，这条问题按已说明关闭；如仍有具体安装错误，欢迎附上操作系统、Python、CUDA、显卡驱动版本以及所用 wheel 文件名重新打开。
